code/z_adventurezones/space_casino.dm

```diff
@@ -0,0 +1,371 @@
+	var/list/homeTables = list() // Initialized early. All nearby tables that the barbuddy should be checking for drinks.
+	var/list/targets = list() // Nearby tables that are in need of drinks.
+	var/target
+	var/atom/moveTowards // The object that should be moved towards.
+
+	var/possible_drinks = list("bilk","beer","cider","mead","wine","champagne","rum","vodka","bourbon", \
+							"boorbon","beepskybeer","screwdriver","bloody_mary","bloody_scary",\
+							"snakebite","diesel","suicider","port","gin","vermouth","bitters","whiskey_sour",\
+							"daiquiri","martini","v_martini","murdini","manhattan","libre","ginfizz","gimlet",\
+							"v_gimlet","w_russian","b_russian","irishcoffee","cosmo","beach","gtonic","vtonic","sonic",\
+							"gpink","eraser","squeeze","hunchback","madmen","planter","maitai","harlow",\
+							"gchronic","margarita","tequini","pfire","bull","longisland","pinacolada","longbeach",\
+							"mimosa","french75","sangria","tomcollins","peachschnapps","moscowmule","tequilasunrise",\
+							"paloma","mintjulep","mojito","cremedementhe","grasshopper","curacao","bluelagoon",\
+							"bluehawaiian","negroni","necroni", "cola", "juice_lime", "juice_lemon", "juice_orange", \
+							"juice_cran", "juice_cherry", "juice_pineapple", "juice_tomato", \
+							"coconut_milk", "sugar", "water", "vanilla", "tea","mint")
+
+	var/possible_poisons = list("acetaldehyde","wolfsbane","ants","weedkiller","cyanide","krokodil", "fluorine", "radium", \
+							"neurotoxin","phlogiston","freeze","dbreath","ghostchilijuice", "cholesterol", "mercury")
+
+	var/possible_vessels = list("/obj/item/reagent_containers/food/drinks/drinkingglass", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/shot", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/oldf", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/round", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/wine", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/cocktail", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/flute")
+
+	var/possible_stuffs = list("/obj/item/cocktail_stuff/drink_umbrella", \
+							"/obj/item/cocktail_stuff/maraschino_cherry", \
+							"/obj/item/cocktail_stuff/cocktail_olive", \
+							"/obj/item/cocktail_stuff/celery")
+
+	var/possible_wedges = list("/obj/item/reagent_containers/food/snacks/plant/orange/wedge", \
+	                        "/obj/item/reagent_containers/food/snacks/plant/lime/wedge", \
+							"/obj/item/reagent_containers/food/snacks/plant/lemon/wedge", \
+							"/obj/item/reagent_containers/food/snacks/plant/grapefruit/wedge")
+
+	New()
+		..()
+		// Start by getting a few initial things
+		if (!src.home || !isturf(src.home))
+			src.home = get_turf(src)
+		if (src.homeTables.len == 0)
+			for (var/obj/table/reinforced/bar/T in view(5, src.home))
+				src.homeTables |= T
```

Comment:
Why that, though? Just curious, cus it seems like it'd prevent any duplicates from being added to the list... even though there shouldnt be any duplicates since its just pulling from view().

Reply:
slower